internal/proxy/proxy.go

```diff
@@ -32,43 +41,61 @@ const (
 
 	// defaultAPTConfigPath is the relative path to the APT proxy configuration file.
 	defaultAPTConfigPath = "etc/apt/apt.conf.d/99ubuntu-proxy-manager"
+
+	// defaultGLibSchemaPath is the relative path to the default GSettings XML schema directory.
+	defaultGLibSchemaPath = "usr/share/glib-2.0/schemas"
+
+	// gschemaOverrideFile is the basename of the GSettings proxy schema override file.
+	gschemaOverrideFile = "99_ubuntu-proxy-manager.gschema.override"
 )
 
 // New returns a new instance of a proxy manager.
 func New(ctx context.Context, args ...option) *Proxy {
 	// Set default options
 	opts := options{
-		root: "/",
+		root:                  "/",
+		glibCompileSchemasCmd: []string{"glib-compile-schemas"},
 	}
 	// Apply given options
 	for _, f := range args {
 		f(&opts)
 	}
 
 	return &Proxy{
-		envConfigPath: filepath.Join(opts.root, defaultEnvConfigPath),
-		aptConfigPath: filepath.Join(opts.root, defaultAPTConfigPath),
+		envConfigPath:       filepath.Join(opts.root, defaultEnvConfigPath),
+		aptConfigPath:       filepath.Join(opts.root, defaultAPTConfigPath),
+		gsettingsConfigPath: filepath.Join(opts.root, filepath.Join(defaultGLibSchemaPath, gschemaOverrideFile)),
```

Comment:
I would assign a variable `glibSchemasPath` just above, directly assign it to it in `glibSchemasPath` and `filepath.Join(glibSchemasPath, gschemaOverrideFile)` to get `gsettingsConfigPath`.

That way, you remove a potential divergence in code using them.